### PR TITLE
fix(cpn): some trainer options not working

### LINKED ISSUE #5676: Trainer Mode Master/Serial turns Off when reading models and settings from radio to Companion
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Companion

### Current Behavior

When I ‘Read Models and Settings from Radio’ to Companion the setting for the ‘Trainer Mode’ changes from ‘Master/Serial’ to ‘Off’.
‘Write Models and Settings to Radio’ from Companion works fine and the Trainer Mode setting is not lost in this transfer.
I noticed this bug in February (version 2.9.4) but didn't report it.

### Expected Behavior

I would expect the Trainer Mode setting on the radio to be transferred to Companion.

### Steps To Reproduce

On a model in the radio, set 'Trainer Mode’ to ‘Master/Serial’.
Open Companion and connect radio to PC.
In Companion click on ‘Read Models and Settings from Radio’.
Open the model.
Check the setting for the ‘Trainer Mode’ and note that it has changed from ‘Master/Serial’ to ‘Off’.

### Version

2.10.5

### Transmitter

FrSky X10 / X10S (ACCST)

### Operating System (OS)

Windows

### OS Version

W11

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pfeerick**
Fixes #5676

Summary of changes:
- allow Master/SBUS Module selection for more radios
- use correct enum for Master/Serial

### CODE REVIEW

File: companion/src/firmwares/edgetx/yaml_modeldata.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ static const YamlLookupTable trainerModeLut = {
   {  TRAINER_MODE_SLAVE_JACK, "SLAVE"  },
   {  TRAINER_MODE_MASTER_SBUS_EXTERNAL_MODULE, "MASTER_SBUS_EXT"  },
   {  TRAINER_MODE_MASTER_CPPM_EXTERNAL_MODULE, "MASTER_CPPM_EXT"  },
-  {  TRAINER_MODE_MASTER_SERIAL, "MASTER_BATT_COMP"  },
+  {  TRAINER_MODE_MASTER_SERIAL, "MASTER_SERIAL"  },
```
Comment (elecpower) [inline companion/src/firmwares/edgetx/yaml_modeldata.cpp:111]: Leave "MASTER_BATT_COMP" in the list for conversion of older settings and add comment "depreciated" but after "MASTER_SERIAL" so encode will write "MASTER_SERIAL"

**Comment by elecpower** [comment]
I too was working on a fix so kudos to you

File: companion/src/firmwares/edgetx/yaml_modeldata.cpp (hunk 1 above)
Comment (pfeerick) [reply companion/src/firmwares/edgetx/yaml_modeldata.cpp:111]: Rodger that... done :)

**Comment by pfeerick** [comment]
np... one less item on that todo list ;) And that makes sense now... I was wondering WTH `MASTER_BATT_COMP` came from but see it was basically renamed some time ago... memory is on the fritz as I see now I wrote the issue about it 🤦‍♂️

**Review by elecpower — APPROVED** [review]

**Pull request merged**